src/modules/data/HashFunction.cpp

```diff
@@ -79,25 +86,22 @@ class MD5 : public HashFunction
 		uint32 c0 = 0x98badcfe;
 		uint32 d0 = 0x10325476;
 
-		//Do the required padding (MD5, SHA1 and SHA2 use the same padding)
-		uint64 paddedLength = length + 1; //Consider the appended bit
-		if (paddedLength % 64 < 56)
-			paddedLength += 56 - paddedLength % 64;
-		if (paddedLength % 64 > 56)
-			paddedLength += 120 - paddedLength % 64;
+		// Compute final padded length, accounting for the appended bit (byte) and size
+		uint64 paddedLength = extend_multiple(length + 1 + 8, 64);
```

Comment:
I think the `+ 8` should be outside of the `extend_multiple` function here (like in the old code), otherwise part of it could be absorbed by the padding even though all 8 bytes need to be available at the end, if I'm understanding right.

Reply:
This function extends/increases the value of `length + 1 + 8` so that it becomes a multiple of 64, thus the result cannot be smaller than `length + 1 + 8` , thus  it cannot be absorbed (if `extend_multiple` is correct).

It is effectively increasing the value by the amount of padding zeroes.

To my understanding, the old code was making `length + 1` congruent to 56 modulo 64, implying `(length + 1) % 64 == 56`, to then add the 8 bytes, making it a multiple of 64.

To clarify, I directly computed the `paddingLength` to be a multiple of 64, rather than thinking about those intermediate computations.

Reply:
Admittedly I don't fully understand the original code, but my assumption would be that the padding exists in the first place so the code can store data there. So if the extra 8 bytes is part of the same input as the original length to produce the padded length, then the algorithm's code might use the extra 8 bytes' memory when it shouldn't (depending on what the original length was), right?

Reply:
Unless I'm missing something, yes this is intended. The final hashed payload is `message + 0x80 + 0x00... + size(8 bytes)` with a length multiple of 64. The old code was adding `8` to `paddedLength` afterwards. It's one of the reasons I removed it, the `paddedLength` was changing progressively, hard to analyze.

Maybe `paddedLength` is a confusing name, but I don't really like `finalLength`.

Reply:
Maybe `payloadLength` would be better ?